vertical-pod-autoscaler/pkg/utils/vpa/capping.go

```diff
@@ -52,12 +52,13 @@ type cappingRecommendationProcessor struct {
 
 // Apply returns a recommendation for the given pod, adjusted to obey policy and limits.
 func (c *cappingRecommendationProcessor) Apply(
-	podRecommendation *vpa_types.RecommendedPodResources,
-	policy *vpa_types.PodResourcePolicy,
-	conditions []vpa_types.VerticalPodAutoscalerCondition,
+	vpa *vpa_types.VerticalPodAutoscaler,
 	pod *apiv1.Pod) (*vpa_types.RecommendedPodResources, ContainerToAnnotationsMap, error) {
 	// TODO: Annotate if request enforced by maintaining proportion with limit and allowed limit range is in conflict with policy.
 
```

Comment:
Can we check for edge cases ( just to avoid panic )?
```go
if vpa == nil {
    return nil, nil, fmt.Errorf("cannot process nil vpa")
    }
if pod == nil {
    return nil, nil, fmt.Errorf("cannot process nil pod")
    }
```